docs/howto/use-extensions/use-the-env-injector-extension.rst

```diff
@@ -0,0 +1,166 @@
+
+Use the env-injector extension
+==============================
+
+The :ref:`env-injector-extension` lets you expose environment variables within the snap
+to users. These variables are accessible by the snap's apps and can modify their
+behavior as they would in a bare host environment.
+
+The user has multiple available methods for setting these environment variables. Your
+snap's apps and design should be oriented toward the most optimal method, and its
+documentation should cover which variables are available and the correct method of
+assigning them.
+
+
+Set up the env-injector extension
+---------------------------------
+
+To add the env-injector extension to an app in your snap:
+
+1. Since env-injector is an `experimental extension
+   <https://snapcraft.io/docs/supported-extensions#p-80380-experimental-extensions>`_,
+   it's blocked by default. To enable experimental extensions during build, your host
+   must set the following environment variable when packing with Snapcraft:
+
+   .. code-block:: bash
+
+       SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=1
+
+2. Your snap must enable configurations with a `configure hook
+   <https://snapcraft.io/docs/supported-snap-hooks#heading--the-configure-hook>`_. If
+   your snap doesn't use this hook yet, it needs at minimum an executable file named
+   ``configure``, without an extension, in ``snap/hooks/``. It should contain at
+   minimum:
+
+   .. code-block:: bash
+
+       #!/bin/sh
+
+       # Optional validation logic
+
+3. In your snap's recipe, the target app's ``extensions`` key must list
+   ``env-injector``. For example, if your snap had an app named ``server``, the key
+   would declare:
+
+   .. code-block:: yaml
+
+       apps:
+         server:
+           command: run.sh
+           daemon: simple
+           extensions: [ env-injector ]
+
+Once set up, the user can set any available environment variables for the snap's apps.
+
+
+Set an environment variable
+---------------------------
+
+When an app in a snap has behavior bound to an environment variable, the user can set it
+either through the `snap's configuration
+<https://snapcraft.io/docs/configuration-in-snaps>`_ or by reading an environment
+(``.env``) file.
+
+Environment variables are applied to apps in one of two ways:
+
+- *Globally*, where the environment variable is passed to all apps that use
+  env-injector.
+- *Locally*, where the environment variable is passed to a specific app that uses
+  env-injector. The app's name is taken from its definition in the snap recipe. The name
+  according to the extension can be :ref:`overridden with an alias
+  <use-the-env-injector-give-app-alias>` to avoid naming conflicts.
+
+
+
+As a snap configuration option
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Users can set environment variables one at a time as a snap configuration options with
+the ``snap set`` command.
+
+To set an environment variable globally, the user can call ``snap set`` and target the
```

Comment:
I'm not sure about the use of "globally" here - that might imply that it works on all snaps, when really it means for all apps in a specific snap.

Reply:
Let me know if this is better: f296e3f